casper/src/main/scala/coop/rchain/casper/api/BlockApi.scala

```diff
@@ -82,6 +85,8 @@ object BlockApi {
   type Error     = String
   type ApiErr[A] = Either[Error, A]
 
+  final class BlockApiException(message: String) extends Exception(message)
```

Comment:
Is there a reason to define specialized exception type?

I'm not against it but before we create clear strategy how to classify errors it can just create confusion that it's defined for some handling reasons.

Reply:
This error class is declared in WebApi.scala, as I wrote in the comment [above](https://github.com/rchain/rchain/pull/3695#discussion_r880596172), but in my opinion it is more logical to move it here.

Reply:
This doesn't explain why this definition is needed. At least it's not used in BlockApi and it's defined in this file.

Please add WebApi changes to this PR also.